io/io/test/TFileMergeClientServer.cxx

```diff
@@ -0,0 +1,84 @@
+#include <memory>
+#include <random>
+#include <thread>
+
+#include "gtest/gtest.h"
+
+#include "TFile.h"
+#include "TROOT.h"
+#include "TTree.h"
+
+#include "ROOT/TFileMergeClient.hxx"
+#include "ROOT/TFileMergeServer.hxx"
```

Comment:
We should sort the includes based on their generality. The file that we test go first, ROOT includes go next, then gtest, then stl. This helps in avoiding to create a non-standalone include files.

For example, if  `TFileMergeClient.hxx` is only used here and it has a `std::thread` in it, it will compile just fine, hiding a bug that we are missing `#include <thread>` in `TFileMergeClient.hxx`